C++/C.sublime-syntax

```diff
@@ -13,1427 +14,773 @@ first_line_match: |-
-      push: line-comment-body
+        1: entity.name.function.c
+        2: punctuation.section.group.begin.c
+      push: [function.tail, function.params]
+    - match: \b{{identifier}}\b
+  function.params:
+    - meta_content_scope: meta.function.parameters.c meta.group.c
+    - match: \)
+      scope: punctuation.section.group.end.c
+      pop: 1
+    - match: \}|\] # Fallback
+      scope: invalid.illegal.stray-bracket-end.c
+      pop: 2
+    - match: ; # Fallback
+      scope: invalid.illegal.unexpected-terminator.c
+      pop: 2
+
+    - match: (?={{identifier}}\s*(?!\s*[\[,\)]|$))
+      push:
+        - match: \b
+          set: function.params.name
+        - include: expression
+    - include: expression
+  function.params.name:
+    - match: (?=[\]\}\);]) # Fallback
+      pop: 1
+    - include: expression-identifiers
+    - match: ({{identifier}})\s*(\[)
+      captures:
+        1: variable.parameter.c
+        2: punctuation.section.brackets.begin.c
+      set: enclosure.brackets
+    - match: ({{identifier}})\s*(?=[,\)]|$)
+      captures:
+        1: variable.parameter.c
+      pop: 1
+    - include: expression
+  function.tail:
+    - meta_scope: meta.function.c
+    - match: ;
+      pop: 1
+    - match: \{
+      scope: punctuation.section.block.begin
+      set: function.body
+    - include: expression
+  function.body:
+    - meta_scope: meta.function.c meta.block.c
+    - match: \}
+      scope: punctuation.section.block.end.c
+      pop: 1
+    - include: expression
+    - include: label
 
-  line-comment-body:
-    - meta_include_prototype: false
-    - meta_scope: comment.line.double-slash.c
-    - include: line-comment-end
+  expression-identifiers:
+    - include: operator.word
+    - include: type
+    - include: keyword
+    - include: constant
 
-  line-comment-documentation-body:
-    - meta_include_prototype: false
-    - meta_scope: comment.line.documentation.c
-    - include: line-comment-end
+###[ SIMPLE STRUCTURES ]############################################################
 
-  line-comment-end:
-    - match: (?:(//+)\s*)?$\n?
+  type:
+    - match: \btypedef\b
+      scope: keyword.declaration.type.c
+      push: type.typedef
+    - match: \benum\b
+      scope: keyword.declaration.enum.c
+      branch_point: enum
+      branch: [type.enum, pop-immediately]
+    - match: \bstruct\b
+      scope: keyword.declaration.struct.c
+      branch_point: struct
+      branch: [type.struct, pop-immediately]
+    - match: \bunion\b
+      scope: keyword.declaration.union.c
+      branch_point: union
+      branch: [type.union, pop-immediately]
+  type.typedef:
+    - match: \(\s*(\*)({{identifier}})\s*\)\s*(\()
+      captures:
+        1: keyword.operator.c
+        2: entity.name.type.typedef.c
+        3: punctuation.section.group.begin.c
+      set: function.params
+
+    - match: ({{identifier}})\s*(?=;)
+      captures:
+        1: entity.name.type.typedef.c
+      pop: 1
+    - match: ; # Fallback
```

Comment:
Missing `scope`? Alternatively use a lookahead and let a parent context consume/scope it, if it is required to handle multiple context pops.